src-tauri/src/lib.rs

```diff
@@ -51,14 +53,41 @@ pub fn get_size<'a>(path: &'a Path) -> Pin<Box<dyn Future<Output = u64> + 'a>> {
     })
 }
 
-pub fn get_file_names(path: &str) -> Vec<String> {
+pub fn get_dir_names(path: &str) -> Vec<String> {
     let mut filenames: Vec<String> = Vec::new();
 
-    for entry in glob_with(path, MatchOptions::default()).expect("glob has no files") {
-        if let Ok(path) = entry {
-            filenames.push(path.display().to_string());
+    let walker = GlobWalkerBuilder::from_patterns(path, &["**/node_modules", "**/![.pnpm]/*"])
+        .max_depth(6)
+        .follow_links(false)
+        .build()
+        .expect("glob walker should work")
+        .into_iter()
+        .filter_map(Result::ok);
+
+    for file in walker {
+        let path_string = file.path().display().to_string();
+        let node_modules_regex = Regex::new(r"node_modules").unwrap();
```

Comment:
One more thing actually, compiling regexes is a (comparatively) expensive operation, plus they can be reused so it makes sense to hoist them out into a global variable (compile it only once at app startup). [Here](https://stackoverflow.com/questions/35169259/how-to-make-a-compiled-regexp-a-global-variable) is a stack overflow showing how.